docs/developer/designs/topology-awareness/README.md

```diff
@@ -0,0 +1,287 @@
+- Inefficient resource utilization across the cluster
+
+## Goals
+
+- Implement topology-aware scheduling in KAI-Scheduler
+- Leverage Kueue's topology CRD concepts to keep an industry standard
+   - Support hierarchical topology representations (e.g., datacenter/rack/node/device)
+- Provide flexible configuration options for topology-based placement
+- Support both Required and Preferred topology constraint request
+
+## Non-Goals
+
+- Automatically detecting topology without user input
+- Ensure backward compatibility with existing scheduling mechanisms
+- Fixing KAI-scheduler scheduling issues with Jobs that use PodAffinity assignments
+
+## Design Details
+
+### Topology Representation
+
+Following Kueue's approach, topology will be represented as a hierarchical structure that describes the physical layout of resources in the cluster.
+We will be using Kueue Topology CRD as is or import its definition into KAI scheduler to simplify integrations and support it as a standard.
+
+### Implementation Approaches
+
+We will consider two main implementation approaches:
+
+#### Approach 1: Topology Tree with resource sums to estimate allocation probability and effect node order by task
+
+Similar to Kueue, this approach involves:
+
+1. Building a tree representation of all topology hierarchies in the cluster
+  * This tree will be updated with each allocation / preemption that is done in simulations
+  * We will need to save and restore it in simulation checkpoints
+2. Summing the resources in each topology tree node
+3. Choosing the topology with the node with the most resources that fits current pod group
+4. Ranking nodes according to their topology grouping
+5. Prioritizing nodes that form the most cohesive topology group for a workload
+6. Considering factors like:
+   - Number of resources available within each topology group
+   - Distance in the tree from the chosen topology group for the Job
+
+
+##### Example of a topology tree structure:
+```mermaid
+graph TD
+    DC[Datacenter: 24 GPUs] --> ZA[Zone-A: 8 GPUs]
+    DC --> ZB[Zone-B: 8 GPUs]
+    DC --> ZC[Zone-C: 8 GPUs]
+
+    ZA --> RA1[Rack-A1: 4 GPUs]
+    ZA --> RA2[Rack-A2: 4 GPUs]
+    ZB --> RB1[Rack-B1: 4 GPUs]
+    ZB --> RB2[Rack-B2: 4 GPUs]
+    ZC --> RC1[Rack-C1: 4 GPUs]
+    ZC --> RC2[Rack-C2: 4 GPUs]
+
+    RA1 --> NA1[Node-A1: 2 GPUs]
+    RA1 --> NA2[Node-A2: 2 GPUs]
+    RA2 --> NA3[Node-A3: 2 GPUs]
+    RA2 --> NA4[Node-A4: 2 GPUs]
+
+    RB1 --> NB1[Node-B1: 2 GPUs]
+    RB1 --> NB2[Node-B2: 2 GPUs]
+    RB2 --> NB3[Node-B3: 2 GPUs]
+    RB2 --> NB4[Node-B4: 2 GPUs]
+
+    RC1 --> NC1[Node-C1: 2 GPUs]
+    RC1 --> NC2[Node-C2: 2 GPUs]
+    RC2 --> NC3[Node-C3: 2 GPUs]
+    RC2 --> NC4[Node-C4: 2 GPUs]
+```
+
+* The distance from node NC4 to NC3 is 2, the distance between RB2 to NA1 is 5.
+* When using hard constraints (required) this will not always find a solution even if one exists in the cluster due to fragmentation of the resources between the nodes
```

Comment:
```suggestion
* When using hard constraints (required) this will not always find a solution even if one exists in the cluster due to fragmentation of the resources between the nodes or other scheduling constraints
```